### PR TITLE
[PETSc] version 3.18.6

### PR DESCRIPTION
**Pull request by boriskaus**
This updates the version of PETSc to 3.18.6

### CODE REVIEW

**Comment by boriskaus** [comment]
hmm, works here but not always on my local machines (that might be a coda-related issue though, see slack discussion).
I will proceed with trying 3.18.5

File: P/PETSc/build_tarballs.jl [hunk 1]
```diff
@@ -136,30 +153,23 @@ build_petsc()
...
+        FOPTFLAGS=${_FOPTFLAGS}  \
         --with-blaslapack-lib=${BLAS_LAPACK_LIB}  \
         --with-blaslapack-suffix=""  \
         CFLAGS='-fno-stack-protector '  \
         FFLAGS="${MPI_FFLAGS}"  \
         LDFLAGS="${LIBFLAGS}"  \
-        FOPTFLAGS='-O3'  \
         --with-64-bit-indices=${USE_INT64}  \
-        --with-debugging=0  \
+        --with-debugging=${DEBUG}  \
```
[the remark this answers is not in the document]
Comment (boriskaus) [reply P/PETSc/build_tarballs.jl:172]: thanks - merged fix

**Comment by boriskaus** [comment]
ok, finally works

File: P/PETSc/build_tarballs.jl [hunk 2]
```diff
@@ -96,56 +109,86 @@ build_petsc()
...
+    LIBFLAGS="-L${libdir}" 
+    if [[ "${target}" == *-mingw* ]]; then
+        LIBFLAGS="-L${libdir} -lssp" 
+    fi
+
+    if [[ "${target}" == *-apple* ]]; then 
+        # on a mac, there might otherwise be confusion with clang
+        CC=gcc
+        FC=gfortran
+        CXX=g++
```
Comment (giordano) [inline P/PETSc/build_tarballs.jl:130]: Not clear to me why we can't use clang, which _is_ what we typically want to use.

Comment (giordano) [reply P/PETSc/build_tarballs.jl]: Sounds like the commit [demonstrate fails when clang is used on apple](https://github.com/JuliaPackaging/Yggdrasil/pull/6533/commits/4d4b46657567d3490d85c344c98f87c3c63bcf55) failed to demonstrate the failure :slightly_smiling_face:

File: P/PETSc/build_tarballs.jl [hunk 3]
```diff
@@ -96,56 +109,86 @@ build_petsc()
...
+    LIBFLAGS="-L${libdir}" 
+    if [[ "${target}" == *-mingw* ]]; then
+        LIBFLAGS="-L${libdir} -lssp" 
+    fi
+
+    if [[ "${target}" == *-apple* ]]; then 
+        # on a mac, there might otherwise be confusion with clang
+      #  CC=gcc
+      #  FC=gfortran
+      #  CXX=g++
```
Comment (giordano) [inline P/PETSc/build_tarballs.jl:130]: This is now dead code, please remove it

File: P/PETSc/build_tarballs.jl (hunk 2 above)
Comment (boriskaus) [reply P/PETSc/build_tarballs.jl]: ok, it did work

File: P/PETSc/build_tarballs.jl [hunk 4]
```diff
@@ -96,56 +109,83 @@ build_petsc()
...
         MUMPS_INCLUDE=""
     fi
+    
+    LIBFLAGS="-L${libdir}" 
+    if [[ "${target}" == *-mingw* ]]; then
+        LIBFLAGS="-L${libdir} -lssp" 
+    fi
+
+    if [[ "${target}" == *-apple* ]]; then 
+        # on a mac, there might otherwise be confusion with clang
```
Comment (giordano) [inline P/PETSc/build_tarballs.jl:127]:
```suggestion
```

**Pull request merged**